make dockerised_all fails

```
$ make dockerised_all_packages 
Logging to docker/log/debian_buster.log
bash: line 7: docker/log/debian_buster.log: No such file or directory
make: *** [docker/docker.mk:116: docker_debian:buster] Error 1
```

Add rule to create directory `docker/log`.

--- a/docker/centos:7.dependencies
+++ b/docker/centos:7.dependencies
@@ -6,3 +6,4 @@ libseccomp-devel
 libsecret-devel
 libsodium-devel
 libsodium-static
+qrencode-devel

--- a/docker/debian:bullseye.dependencies
+++ b/docker/debian:bullseye.dependencies
@@ -3,6 +3,7 @@ help2man
 libcjson-dev
 libcurl4-openssl-dev
 libmicrohttpd-dev
+libqrencode-dev
 libseccomp-dev
 libsecret-1-dev
 libsodium-dev

--- a/docker/docker-build.sh
+++ b/docker/docker-build.sh
@@ -63,6 +63,17 @@ debian_copy_output() {
     mv ../${PACKAGE_DIR}[_-]* $OUTPUT/$DIST
     mv ../lib* $OUTPUT/$DIST
 }
+
+rpm_unfuck_source_tarball_location() {
+	cat rpm/oidc-agent.spec \
+        | grep -v ^Source \
+        > rpm/oidc-agent.spec.bckp
+    VERSION=`head debian/changelog -n 1|cut -d \( -f 2|cut -d \) -f 1|cut -d \- -f 1`
+    RELEASE=`head debian/changelog -n 1|cut -d \( -f 2|cut -d \) -f 1|cut -d \- -f 2`
+    sed "s/#DO_NOT_REPLACE_THIS_LINE/Source0: oidc-agent-${VERSION}.tar.gz/" -i rpm/oidc-agent.spec.bckp
+    rm -f rpm/oidc-agent.spec
+    mv rpm/oidc-agent.spec.bckp rpm/oidc-agent.spec
+}
 rpm_build_package() {
     cd /tmp/build/$PACKAGE_DIR
     make rpmsource
@@ -97,14 +108,15 @@ case "$DIST" in
         debian_copy_output
     ;;
     centos_8|centos_7|fedora*)
+        rpm_unfuck_source_tarball_location
         rpm_build_package
         rpm_copy_output
     ;;
     opensuse_15*|opensuse_tumbleweed|sle*)
+        rpm_unfuck_source_tarball_location
         rpm_build_package
         rpm_copy_output
     ;;
 esac
 
 common_fix_output_permissions
-bash

--- a/docker/docker.mk
+++ b/docker/docker.mk
@@ -113,6 +113,7 @@ dockerised_deb_debian_buster: docker_debian\:buster
 .PHONY: docker_debian\:buster
 docker_debian\:buster:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -129,6 +130,7 @@ dockerised_deb_debian_bullseye: docker_debian\:bullseye
 .PHONY: docker_debian\:bullseye
 docker_debian\:bullseye:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -145,6 +147,7 @@ dockerised_deb_debian_bookworm: docker_debian\:bookworm
 .PHONY: docker_debian\:bookworm
 docker_debian\:bookworm:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -162,6 +165,7 @@ dockerised_deb_ubuntu_bionic: docker_ubuntu\:bionic
 .PHONY: docker_ubuntu\:bionic
 docker_ubuntu\:bionic:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -178,6 +182,7 @@ dockerised_deb_ubuntu_focal: docker_ubuntu\:focal
 .PHONY: docker_ubuntu\:focal
 docker_ubuntu\:focal:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -194,6 +199,7 @@ dockerised_deb_ubuntu_hirsute: docker_ubuntu\:hirsute
 .PHONY: docker_ubuntu\:hirsute
 docker_ubuntu\:hirsute:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -210,6 +216,7 @@ dockerised_deb_ubuntu_impish: docker_ubuntu\:impish
 .PHONY: docker_ubuntu\:impish
 docker_ubuntu\:impish:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_APT_INIT)"\n" \
@@ -228,6 +235,7 @@ dockerised_rpm_centos_7: docker_centos\:7
 .PHONY: docker_centos\:7
 docker_centos\:7: 
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_YUM_DISABLE_FASTESTMIRROR)"\n" \
@@ -247,6 +255,7 @@ dockerised_rpm_centos_8: docker_centos\:8
 .PHONY: docker_centos\:8
 docker_centos\:8: 
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_YUM_BUILD_ESSENTIALS)"\n" \
@@ -266,6 +275,7 @@ dockerised_rpm_opensuse_15.2: docker_registry.opensuse.org/opensuse/leap\:15.2
 .PHONY: docker_registry.opensuse.org/opensuse/leap\:15.2
 docker_registry.opensuse.org/opensuse/leap\:15.2:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_ZYP_BUILD_ESSENTIALS)"\n" \
@@ -282,6 +292,7 @@ dockerised_rpm_opensuse_15.3: docker_registry.opensuse.org/opensuse/leap\:15.3
 .PHONY: docker_registry.opensuse.org/opensuse/leap\:15.3
 docker_registry.opensuse.org/opensuse/leap\:15.3:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_ZYP_BUILD_ESSENTIALS)"\n" \
@@ -298,6 +309,7 @@ dockerised_rpm_opensuse_tumbleweed: docker_registry.opensuse.org/opensuse/tumble
 .PHONY: docker_registry.opensuse.org/opensuse/tumbleweed\:latest
 docker_registry.opensuse.org/opensuse/tumbleweed\:latest:
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_ZYP_BUILD_ESSENTIALS)"\n" \
@@ -315,6 +327,7 @@ dockerised_rpm_fedora_34: docker_fedora\:34
 .PHONY: docker_fedora\:34
 docker_fedora\:34: 
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	@echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_YUM_BUILD_ESSENTIALS)"\n" \
@@ -332,6 +345,7 @@ dockerised_rpm_fedora_35: docker_fedora\:35
 .PHONY: docker_fedora\:35
 docker_fedora\:35: 
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_YUM_BUILD_ESSENTIALS)"\n" \
@@ -349,6 +363,7 @@ dockerised_rpm_fedora_rawhide: docker_fedora\:rawhide
 .PHONY: docker_fedora\:rawhide
 docker_fedora\:rawhide: 
 	@echo Logging to ${DOCKER_LOG}
+	@test -d docker/log || mkdir -p docker/log
 	echo -e \
 	$(DOCKER_GEN_FROM_IMAGE)"\n" \
 	$(DOCKER_YUM_BUILD_ESSENTIALS)"\n" \

--- a/docker/fedora:34.dependencies
+++ b/docker/fedora:34.dependencies
@@ -6,3 +6,4 @@ libseccomp-devel
 libsecret-devel
 libsodium-devel
 libsodium-static
+qrencode-devel

--- a/docker/opensuse_leap:15.2.dependencies
+++ b/docker/opensuse_leap:15.2.dependencies
@@ -6,3 +6,4 @@ libseccomp-devel
 libsecret-devel
 libsodium-devel
 libsodium23
+qrencode-devel

--- a/docker/opensuse_leap:15.2.dependencies
+++ b/docker/opensuse_leap:15.2.dependencies
@@ -7,3 +7,4 @@ libsecret-devel
 libsodium-devel
 libsodium23
 qrencode-devel
+unzip

--- a/docker/ubuntu:bionic.dependencies
+++ b/docker/ubuntu:bionic.dependencies
@@ -2,6 +2,7 @@ check
 help2man
 libcurl4-openssl-dev
 libmicrohttpd-dev
+libqrencode-dev
 libseccomp-dev
 libsecret-1-dev
 libsodium-dev

--- a/docker/ubuntu:focal.dependencies
+++ b/docker/ubuntu:focal.dependencies
@@ -2,6 +2,7 @@ check
 help2man
 libcurl4-openssl-dev
 libmicrohttpd-dev
+libqrencode-dev
 libseccomp-dev
 libsecret-1-dev
 libsodium-dev

--- a/rpm/oidc-agent.spec
+++ b/rpm/oidc-agent.spec
@@ -12,8 +12,8 @@ URL: https://github.com/indigo-dc/oidc-agent
 # use `make rpmsource` to generate the required tarball
 #Source0: https://github.com/indigo-dc/oidc-agent/archive/refs/heads/master.zip
 #Source0: https://github.com/indigo-dc/oidc-agent/archive/refs/heads/docker-builds.zip
-#Source0: oidc-agent-4.1.1.tar.gz
 Source0: https://github.com/indigo-dc/oidc-agent/archive/refs/tags/v%{version}.tar.gz
+#DO_NOT_REPLACE_THIS_LINE
 
 BuildRequires: gcc >= 4.8
 BuildRequires: libcurl-devel >= 7.29
@@ -31,6 +31,7 @@ BuildRequires: libseccomp-devel >= 2.3
 BuildRequires: help2man >= 1.41
 BuildRequires: libsecret-devel >= 0.18.4
 BuildRequires: desktop-file-utils
+BuildRequires: qrencode-devel
 
 Requires: oidc-agent-desktop == %{version}-%{release}
 BuildRoot:	%{_tmppath}/%{name}

